src/atom.rs

```diff
@@ -32,6 +41,38 @@ impl NifAtom {
             false => None
         }
     }
+
+    /// Get the contents of this atom as a string.
+    ///
+    /// (This is slow, and for most cases, unnecessary. For example, if you
+    /// need to know whether or not this atom is `true`, try comparing it to
+    /// `get_atom_init("true")`.)
+    ///
+    pub fn to_string(&self, env: &NifEnv) -> String {
```

Comment:
My implementation of this is a bit different. It is done completely without additional copying, but it allocates a full 257 bytes of memory regardless of how long the atom actually is. What do you think?

```rust
pub fn get_atom(env: NIF_ENV, term: NIF_TERM) -> Option<String> {
    // Max 257 characters can be written (256 + NULL).
    let mut string = String::with_capacity(257);

    {
        // We will validate UTF8 at the end of the function.
        let mut string_buf = unsafe { string.as_mut_vec() };

        let result = unsafe { nif_interface::enif_get_atom_latin1(env, term, string_buf.as_mut_ptr(), 257) };

        if result == 0 {
            return None;
        }

        // This SHOULD be safe unless the VM is lying to us.
        unsafe { string_buf.set_len((result - 1) as usize) };
    }

    // REQUIRED! This will verify that the data is actually valid
    // unicode without copying.
    match ::std::str::from_utf8(string.as_bytes()) {
        Ok(_) => Some(string),
        Err(_) => None,
    }
}
```

Note that my variant is untested, but it should mostly work.